finagle-netty4/src/main/scala/com/twitter/finagle/netty4/transport/ChannelTransport.scala

```diff
@@ -139,7 +139,7 @@ private[finagle] class ChannelTransport(ch: Channel) extends Transport[Any, Any]
     closed.unit
   }
 
-  val peerCertificate: Option[Certificate] = ch.pipeline.get(classOf[SslHandler]) match {
+  def peerCertificate: Option[Certificate] = ch.pipeline.get(classOf[SslHandler]) match {
```

Comment:
I guess when we'll actually delay the `ChannelTransport` creation, the peer certificate will be available at the moment this constructor fires and yet I still think it should be `def`, not `val`.  I like this more by two reasons.

1. I'd imagine that sessions could be renegitoated so having `def` means we could get the latest peer certificate if that happens.
2. This would reduce the memory footprint for each channel-transport (each connection).

Reply:
hello,

was this merge? i don't see it on 18.1.0 nor on develop